### PR TITLE
Counts the number of leaf nodes in a binary search tree in python

### LINKED ISSUE #4448: Leaf nodes in a bst
## 🚀 Feature

To count the number of leaf nodes in a binary search tree in Python.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Pitch
Binary search tree is an important data structure and leaf nodes can be counted using recursion

### PR DESCRIPTION
**Pull request by aish2002**
Fixes #4448 

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Counts the number of leaf nodes in a binary search tree

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#4448

### CODE REVIEW

**Comment by aish2002** [comment]
@HarshCasper , @siddharth25pandey , @ankitaggarwal23  kindly review the PR

**Review by gat786 — APPROVED** [review]

File: Python/ds/Leaf_nodes_BST.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        10
+        1
+        4
+        8
+        11
+    Sample output:
+        No. of leaf nodes in the binary search tree = 4
+
+    Time complexity : O(n)
+    Space complexity : O(n)
```
Comment (ankitaggarwal23) [inline Python/ds/Leaf_nodes_BST.py]: Remove the spaces at the start

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Rest looks good to merge

File: Python/ds/Leaf_nodes_BST.py (hunk 1 above)
Comment (aish2002) [reply Python/ds/Leaf_nodes_BST.py]: Removed the spaces. Kindly re-review the PR

File: Python/ds/Leaf_nodes_BST.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    print("Enter the numbers: ")
+    for _ in range(0, n):
+        data = int(input())
+        root = insertIntoTree(root, data)
+    # Initially count of leaf nodes is set as 0
+    leaf_nodes.count = 0
+    print("\nNo. of leaf nodes in the binary search tree = " +str(leaf_nodes(root)))
+
+if __name__ == "__main__":
+    main()
```
Comment (ankitaggarwal23) [inline Python/ds/Leaf_nodes_BST.py:62]: Add a new line after main()

File: Python/ds/Leaf_nodes_BST.py (hunk 1 above)
Comment (ankitaggarwal23) [reply Python/ds/Leaf_nodes_BST.py]: Removing spaces means,
Capture

Write it like that!

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Python/ds/Leaf_nodes_BST.py (hunk 1 above)
Comment (aish2002) [reply Python/ds/Leaf_nodes_BST.py]: Made the changes. Kindly re-review the code

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by HarshCasper** [comment]
Please make an issue here and raise a PR, for an article as part of our Docs repository: https://github.com/TesseractCoding/NeoAlgo-Docs

It would be much appreciated. Thanks!

**Pull request merged**